src/cat.c

```diff
@@ -72,6 +72,9 @@ static void reset_state(struct cat_object *self)
         }
         self->cmd = NULL;
         self->cmd_type = CAT_CMD_TYPE_NONE;
+        #if CONFIG_LIB_CAT_USRDATA
+        cat_user_data_init();
```

Comment:
calling functions from external cAT-related modules needs to pass current context (self pointer). it allows to continue object-oriented architecture.